### PR TITLE
KTO: Replace _get_train_sampler with train_sampling_strategy for transformers >= 5.2.0

### PR DESCRIPTION
**Pull request by albertvillanova**
Align KTO with DPO: Replace `_get_train_sampler` with `train_sampling_strategy`.

This PR updates the KTO configuration and trainer to enforce and document the requirement for sequential sampling during training, which is necessary for correct KL computation. It also cleans up unused imports and removes an unused method.

Part of:
- #4786

### Motivation

`KTOTrainer` previously overrode `_get_train_sampler` to unconditionally return a `SequentialSampler`, ignoring whatever `train_sampling_strategy` was set to. The sequential order is required because the KL completion for each example is precomputed against its neighbors in a fixed-order batch (see `_get_kl_dataset`): shuffling the training order would break that pairing and silently degrade KL estimation.

Using `train_sampling_strategy` is the standard mechanism for this in the `Trainer` base class. The override is unnecessary once the default is set correctly.

### Changes

**KTO sequential sampling enforcement and documentation:**

* Updated the `KTOConfig` class to add a `train_sampling_strategy` parameter, defaulting to `"sequential"` (overriding the `TrainingArguments` default of `"random"`), and clarified in the documentation why sequential sampling is required for KTO.
* In `KTOTrainer.__init__`, added a runtime check to raise a `ValueError` if `train_sampling_strategy` is not set to `"sequential"`, enforcing the requirement programmatically.

**Code cleanup:**

* Removed the now-unnecessary `_get_train_sampler` method from `KTOTrainer`, now redundant, since the base class `Trainer._get_train_sampler` already returns a `SequentialSampler` when `train_sampling_strategy == "sequential"`.
* Removed unused imports: `SequentialSampler` and `has_length` from `kto_trainer.py`.

### Breaking change

> [!NOTE]
> Previously, setting `train_sampling_strategy` to any non-sequential value had no effect: the override silently ignored it.
> After this change, passing `train_sampling_strategy="random"` or `"group_by_length"` (including via a plain `TrainingArguments` instance, whose default is `"random"`) raises a `ValueError`
>
> Users who were unknowingly passing a non-sequential strategy (with no visible effect) will need to either remove the override or set `train_sampling_strategy="sequential"` explicitly.

---

> [!NOTE]
> **Medium Risk**
> Behavior change: configs with non-sequential `train_sampling_strategy` (including plain `TrainingArguments` defaults) now fail fast instead of being overridden, which can break existing scripts that relied on the old silent override.
> 
> **Overview**
> KTO training is aligned with the DPO-style **`train_sampling_strategy`** knob instead of silently forcing a sequential sampler in all cases.
> 
> **`KTOConfig`** now defaults **`train_sampling_strategy`** to **`"sequential"`** (overriding generic training defaults) and documents that KTO needs fixed batch order so precomputed KL completions stay paired with their neighbors. **`KTOTrainer`** raises **`ValueError`** at init if any other strategy is set, so non-sequential configs are no longer ignored.
> 
> **`_get_train_sampler`** is narrowed: on **Transformers &lt; 5.2.0** it still returns **`SequentialSampler`**; on newer versions it defers to the base **`Trainer`** implementation driven by **`train_sampling_strategy`** (which must remain sequential).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit fd2b5bd982c2220de3a36712bdcdd31de7340f75. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/kto/kto_trainer.py [hunk 1]
```diff
@@ -343,6 +343,12 @@ def __init__(
         self.aux_loss_enabled = getattr(model.config, "output_router_logits", False)
         self.aux_loss_coef = getattr(model.config, "router_aux_loss_coef", 0.0)
         self.calculate_KL = False if self.loss_type in ["apo_zero_unpaired"] else True
+        if args.train_sampling_strategy != "sequential":
+            raise ValueError(
+                f"KTO requires `train_sampling_strategy='sequential'` because the KL completion for each example is "
+                f"precomputed against its neighbors in a fixed-order batch; any other strategy breaks that pairing. "
+                f"Got `train_sampling_strategy='{args.train_sampling_strategy}'`."
+            )
```
[the remark this answers is not in the document]
Comment (albertvillanova) [reply trl/experimental/kto/kto_trainer.py:351]: I am checking this.

**Review by qgallouedec — APPROVED** [review]
Nice!

NIT it'd drop "Align KTO with DPO" in the title

File: trl/experimental/kto/kto_trainer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/experimental/kto/kto_trainer.py:351]: Ah I'm afraid this is correct :/

[the remark this answers is not in the document]
Comment (albertvillanova) [reply trl/experimental/kto/kto_trainer.py:351]: Cursor is right: the possibility to get a `SequentialSampler` from `Trainer._get_train_sampler` with `train_sampling_strategy` was introduced in transformers v5.2.0 by:
- https://github.com/huggingface/transformers/pull/43138

:sweat:

**Comment by albertvillanova** [comment]
I'm afraid this PR could only be merged when we require `transformers` >= 5.2.0.

I'm closing it for now.

**Comment by albertvillanova** [comment]
I'm reopening this to make a proposal: make the dependency on `transformers` explicit, so we can easily update it once we drop transformers < 5.2.0.

Additionally, I have a follow-up PR to support non-sequential Sampler with "apo_zero_unpaired" loss type (`calculate_KL=False`).

**Review by qgallouedec — APPROVED** [review]
yep, I think this is the best way at this point
